### PR TITLE
feat: Use self-signed certs for TLS (between components and between k8s and KEDA)

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

This PR adds support for using TLS between KEDA components with client authentication,not just for encrypting the traffic, the client MUST authenticate itself using its own certificate. This reduces the risk of exposing metrics ignoring the cluster RBAC (if the operator grpc server doesn't have authentication, everyone could connect to it and get metric values, ignoring the cluster RBAC required for querying the metrics server).
This PR also register the injected certificates in the metrics server and exposes the tls cert instead of self-generating it on every restart. Now using the injected certificate (from our self-signed cert), we can (and we do) patch the apiservice manifest to include the caBundle, removing `insecureSkipTLSVerify` and trusting in the CA

## Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/3931

### CODE REVIEW

File: pkg/certificates/certificate_manager.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	secrets := &corev1.SecretList{}
+	kedaNamespace := kedautil.GetPodNamespace()
+	opt := &client.ListOptions{
+		Namespace: kedaNamespace,
+	}
+
+	err := mgr.GetAPIReader().List(ctx, secrets, opt)
+	if err != nil {
+		cm.Logger.Error(err, "unable to check secrets")
+		os.Exit(1)
```
Comment (zroubalik) [inline pkg/certificates/certificate_manager.go:123]: let's return error here instead

File: cmd/adapter/main.go [hunk 2]
```diff
@@ -200,7 +199,7 @@ func runScaledObjectController(ctx context.Context, mgr manager.Manager, scaleHa
 // generateDefaultMetricsServiceAddr generates default Metrics Service gRPC Server address based on the current Namespace.
 // By default the Metrics Service gRPC Server runs in the same namespace on the keda-operator pod.
 func generateDefaultMetricsServiceAddr() string {
-	return fmt.Sprintf("keda-operator.%s.svc.cluster.local:9666", kedautil.GetPodNamespace())
```
[the remark this answers is not in the document]
Comment (JorTurFer) [reply cmd/adapter/main.go:203]: Does using the full name make sense? The minimum required is service.namespace.
I can revert this change and add this full name to the cert if you prefer it

File: pkg/certificates/certificate_manager.go (hunk 1 above)
Comment (JorTurFer) [reply pkg/certificates/certificate_manager.go:123]: f**k, another copy-pasting error, nice catch

Comment (JorTurFer) [reply pkg/certificates/certificate_manager.go]: updated!

File: cmd/adapter/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zroubalik) [reply cmd/adapter/main.go:203]: I am not sure, but I prefer the full name. Sometimes there are problems with discovery and once we have the mutli tenant solution, there might be problems, maybe not. But I think that the full name doesn't cause any problems, right?

[the remark this answers is not in the document]
Comment (JorTurFer) [reply cmd/adapter/main.go:203]: you're right. I'm gonna revert this change and include this dns name to the cert, at the end of the day, it's free xD

**Review by zroubalik — APPROVED** [review]
Great work!

**Pull request merged**